modules/apps/change-tracking/change-tracking-test/src/testIntegration/java/com/liferay/change/tracking/internal/test/PublicationUserNotificationHandlerTest.java

```diff
@@ -272,6 +261,176 @@ public void testGetLinkToViewConflicts() throws Exception {
 		}
 	}
 
+	@Test
+	public void testGetLinkToViewStackTrace() throws Exception {
+		CTCollection ctCollection =
+			CTCollectionTestUtil.createCTCollectionwithJournalArticle();
+
+		Bundle journalServiceBundle = null;
+
+		try {
+			Bundle bundle = FrameworkUtil.getBundle(
+				JournalArticleService.class);
+
+			BundleContext bundleContext = bundle.getBundleContext();
+
+			for (Bundle curBundle : bundleContext.getBundles()) {
+				if (Objects.equals(
+						curBundle.getSymbolicName(),
+						"com.liferay.journal.service") &&
+					(curBundle.getState() == Bundle.ACTIVE)) {
+
+					curBundle.stop();
+
+					journalServiceBundle = curBundle;
+
+					break;
+				}
+			}
+
+			CTCollectionTestUtil.publishCTCollectionWithError(
+				ctCollection.getCtCollectionId());
+
+			List<UserNotificationEvent> userNotificationEvents =
+				_userNotificationEventLocalService.getUserNotificationEvents(
+					TestPropsValues.getUserId());
+
+			for (UserNotificationEvent userNotificationEvent :
+					userNotificationEvents) {
+
+				if (!Objects.equals(
+						CTPortletKeys.PUBLICATIONS,
+						userNotificationEvent.getType())) {
+
+					continue;
+				}
+
+				JSONObject jsonObject = _jsonFactory.createJSONObject(
+					userNotificationEvent.getPayload());
+
+				if ((jsonObject.getInt("notificationType") !=
+						UserNotificationDefinition.
+							NOTIFICATION_TYPE_REVIEW_ENTRY) ||
+					jsonObject.getBoolean("showConflicts") ||
+					!Objects.equals(
+						userNotificationEvent.getUserId(),
+						_getAdminUserId(
+							userNotificationEvent.getCompanyId()))) {
+
+					continue;
+				}
+
+				ServiceContext serviceContext = _getServiceContext();
+
+				UserNotificationFeedEntry userNotificationFeedEntry =
+					_userNotificationHandler.interpret(
+						userNotificationEvent, serviceContext);
+
+				Assert.assertEquals(
+					userNotificationFeedEntry.getLink(),
+					PortletURLBuilder.create(
+						_portal.getControlPanelPortletURL(
+							serviceContext.getRequest(),
+							serviceContext.getScopeGroup(),
+							CTPortletKeys.PUBLICATIONS, 0, 0,
+							PortletRequest.RENDER_PHASE)
+					).setMVCRenderCommandName(
+						"/change_tracking/view_stack_trace"
+					).setParameter(
+						"backgroundTaskId",
+						_jsonFactory.createJSONObject(
+							userNotificationEvent.getPayload()
+						).getLong(
+							"backgroundTaskId"
+						)
+					).buildString());
+			}
+		}
+		finally {
+			if (journalServiceBundle != null) {
+				journalServiceBundle.start();
+			}
+		}
+	}
+
+	@Test
+	public void testGetUserNotificationEvent() throws Exception {
```

Comment:
Isn't this test almost the same as `testGetBodyForStoppedService`? Let's update the other test to cover regular publication user and the admin.